packages/extension-utils/src/OAuthConfigForm/OAuthConfigForm.tsx

```diff
@@ -0,0 +1,400 @@
+  configKey: string
+  /** OAuth client id */
+  clientId: string
+  /** OAuth client display label */
+  clientLabel: string
+}
+
+export const OAuthConfigForm = ({
+  setHasConfig,
+  clientId,
+  clientLabel,
+  configKey,
+}: ConfigFormProps) => {
+  const location = useLocation()
+  const redirect_uri = appPath(location, '/oauth')
+  const BASE_URL = 'baseUrl'
+  const WEB_URL = 'webUrl'
+  const FETCH_INTENT = 'fetchIntent'
+  const FETCH_RESULT = 'fetchResult'
+  const CRITICAL: MessageBarIntent = 'critical'
+  const appConfig = `// client_guid=${clientId}
+{
+  "redirect_uri": "${redirect_uri}",
+  "display_name": "CORS ${clientLabel}",
+  "description": "Looker ${clientLabel} using CORS",
+  "enabled": true
+}
+`
+  const adaptor = getEnvAdaptor()
+  const sdk = adaptor.sdk
+  // See https://codesandbox.io/s/youthful-surf-0g27j?file=/src/index.tsx for a prototype from Luke
+  // TODO see about useReducer to clean this up a bit more
+
+  const getConfig = () => {
+    // TODO: This is temporary until config settings are available in redux.
+    // get configuration from storage, or default it
+    const data = localStorage.getItem(configKey)
+    const result = data ? JSON.parse(data) : EmptyConfig
+    return result
+  }
+
+  const config = getConfig()
+  const [fields, setFields] = useState<IFieldValues>(defaultFieldValues)
+  const [saved, setSaved] = useState<ConfigValues>(config)
+
+  const updateFields = (
+    nameOrValues: string | Partial<IFieldValues>,
+    value = ''
+  ) => {
+    if (typeof nameOrValues === 'string') {
+      setFields((previousFields) => {
+        return { ...previousFields, ...{ [nameOrValues]: value } }
+      })
+    } else {
+      setFields((previousFields) => {
+        return { ...previousFields, ...nameOrValues }
+      })
+    }
+  }
+
+  useEffect(() => {
+    const data = getConfig()
+    const { base_url, looker_url } = data
+    setSaved(data)
+    updateFields({
+      [BASE_URL]: base_url,
+      [WEB_URL]: looker_url,
+      [FETCH_INTENT]:
+        base_url !== '' && looker_url !== '' ? POSITIVE : CRITICAL,
+    })
+  }, [])
+
+  const [validationMessages, setValidationMessages] =
+    useState<ValidationMessages>({})
+
+  const updateMessage = (intent: MessageBarIntent, message: string) => {
+    updateFields({ [FETCH_RESULT]: message, [FETCH_INTENT]: intent })
+  }
+
+  const isConfigured = () => {
+    return (
+      saved !== EmptyConfig &&
+      fields[BASE_URL] === saved.base_url &&
+      fields[WEB_URL] === saved.looker_url
+    )
+  }
+
+  const fetchError = (message: string) => {
+    updateFields(WEB_URL, '')
+    updateMessage(CRITICAL, message)
+  }
+
+  const saveConfig = (baseUrl: string, webUrl: string) => {
+    const data = {
+      base_url: baseUrl,
+      looker_url: webUrl,
+      client_id: clientId,
+      redirect_uri,
+    }
+    updateFields({
+      [BASE_URL]: baseUrl,
+      [WEB_URL]: webUrl,
+    })
+    // TODO: replace when redux is introduced
```

Comment:
Even when we have redux, I think we'd still want to save OAuth configuration to local storage as well, so it can be used for the restart source

Reply:
Good deal!